db/db.go

```diff
@@ -166,6 +166,32 @@ func (db *DB) Get(caller Caller, name string) (*api.SecretValue, error) {
 	return db.kv.get(name)
 }
 
+// GetConditional returns a secret's active value if it is different from oldVersion.
+// If the active version is the same as oldVersion, it reports api.ErrValueNotChanged.
+func (db *DB) GetConditional(caller Caller, name string, oldVersion api.SecretVersion) (*api.SecretValue, error) {
+	// This case is special in that we only log an access if the condition
+	// succeeds and we report a fresh value to the caller. However, we still
+	// want a log if authorization fails.
+	if !caller.Permissions.Allow(acl.ActionGet, name) {
```

Comment:
I thought about trying to pull apart checkAndLog, but the permissions check is the only thing that's really separate, and it's not very expensive to repeat so I didn't bother.